consumer_base.go

```diff
@@ -126,6 +126,7 @@ func (c *base) startConsume() {
 	for {
 		select {
 		case <-c.quit:
+			close(c.incomingMessageStream)
```

Comment:
Before, we had closed this channel in `Stop()` method, but this is not the correct way to manage channels. 

If a goroutine writes to a channel, It should be that goroutine's responsibility to close the channel, too.